### PR TITLE
 Do not use typedefs that are not emitted.

### PR DESCRIPTION
**Pull request by mprobst**
`skipEmitRegExp` would lead to typedefs that are not emitted being referenced by code that is emitted.

This also contains a number of minor fixups I made on the way.

### CODE REVIEW

**Comment by mprobst** [comment]
G3 is green.

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java [hunk 1]
```diff
@@ -337,9 +337,11 @@ void collectTypedefs() {
       if (type == null
```
Comment (evmar) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: Better first line for this commit: "Minor syntax fixes".  ("Minor fixes" is too vague, hard to know if it's fixing minor clutz bugs or some other thing.)

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java [hunk 2]
```diff
@@ -683,8 +700,7 @@ private void processUnprovidedTypes(Set<String> provides, Set<String> transitive
         if (symbolInput != null && symbolInput.isExtern()) continue;
 
         // skip types that come from files that we skip emit on.
```
Comment (evmar) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: This comment now reads like it's peering into the impl of shouldSkipVar.  I'd say maybe just drop it.

File: .gitignore [hunk 3]
```diff
@@ -15,5 +15,8 @@ out/
 *.idea
```
Comment (evmar) [inline .gitignore]: Commit title: "*git* ignore [...]" (thought it was talking about making clutz ignore it somehow)

Comment (mprobst) [reply .gitignore]: Done.

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java (hunk 1 above)
Comment (mprobst) [reply src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: Done.

**Pull request merged**